R_program/Main.R

```diff
@@ -553,6 +566,181 @@ remove(temp)
 
 ############################# End MARKOV Weather Generator Code ##############################
 
+############################# Vegetation Code ##############################
+# only rescale space if requested.
+if(rescale_space){
+  # This code determines plant functional type relative abundance/composition
+  # and then scales STEPWAT2 parameters accordingly
+  source(vegetation.file)
+
+  # Array of plant functional type relative abundance/composition
+  relVegAbund <- estimate_STEPWAT_relativeVegAbundance(sw_weatherList)
+
+  # vectors that map rgroup names to the columns names of relVegAbund
+  Shrubs <- c("sagebrush", "shrub")
+  Forbs <- c("a.cool.forb","a.warm.forb", "p.cool.forb", "p.warm.forb")
+  Succulents <- c("succulents")
+  Grasses_C3 <- c("p.cool.grass")
+  Grasses_C4 <- c("p.warm.grass")
+  Grasses_Annuals <- c("a.cool.grass", "a.warm.grass")
+  Trees <- c()
+
+  # move to dist folder so we can begin adjusting space
+  setwd("../STEPWAT_DIST")
+
+  # will store the new rgroup files temporarily
+  new_rgroup_files <- c()
+
+  # Loop through all of the rgroup files defined in inputs
+  for(rg in rgroups){
+    #read the start of the file (where space is defined)
+    rgrp <- readLines(con <- paste0(rg,".in"), n = 10)
+    # split the file along tabs. This produces a 2d array of entries where rows are lines of the original file
+    # and columns are the entries of each line
+    rgrp <- strsplit(rgrp, "\t")
+    
+    # These vectors store the space parameters already in the rgroup file.
+    # They are needed in case one functional type has two rgroup entries.
+    shrub_space <- c(); forb_space <- c(); succulent_space <- c(); c3_space <- c()
+    c4_space <- c(); annuals_space <- c(); tree_space <- c()
+  
```

Comment:
@chaukap if some of these functional types are turned off, the code will still work as expected correct?

Reply:
@kpalmqui I believe trees are already turned off. The math behind it works regardless of how many functional types are used, and coupled with R's floating point rounding, the algorithm should produce the correct output even if there is only one functional type.